.travis.yml

```diff
@@ -5,14 +5,34 @@ go:
 - 1.7.x
 - 1.8.x
 - 1.9.x
+- 1.x
 
 install:
   - go get -v -d -t github.com/golang/protobuf/...
   - curl -L https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip -o /tmp/protoc.zip
   - unzip /tmp/protoc.zip -d $HOME/protoc
+  - mkdir -p $HOME/src && ln -s $HOME/protoc $HOME/src/protobuf
 
 env:
   - PATH=$HOME/protoc/bin:$PATH
 
 script:
-  - make all test
+  - make all
+  - make regenerate
```

Comment:
This doesn't seem right--if we run regenerate before test, the golden tests will always pass even if something has unexpectedly changed.

Reply:
I'm happy to rearrange this, but note that if anything at all changes as a result of `make regenerate`, the stanza immediately following will fail.

Reply:
Ahh, I see what you're doing; I hadn't quite followed the purpose of that stanza.

FWIW, the tests in protoc-gen-go should already pass independent of the Go version (they ignore the compressed portion of the generated files). Would it make more sense to fix any of the tests which don't have that behavior than to change the Travis config?

Reply:
> FWIW, the tests in protoc-gen-go should already pass independent of the Go version (they ignore the compressed portion of the generated files). Would it make more sense to fix any of the tests which don't have that behavior than to change the Travis config?

Again, while those tests might pass, the files will have changed and the aforementioned stanza will fail.

Reply:
Also note that no tests are being skipped here; `make test` is run unconditionally.

Reply:
I guess my question is: If this is doing something that `make test` doesn't already do, should we just fix `make test`?

("Fixing `make test` is hard" is a valid answer. :) )